src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyPairGenerator.java

```diff
@@ -99,29 +99,35 @@
         // set default key sizes and apply our own algorithm-specific limits
         // override lower limit to disallow unsecure keys being generated
         // override upper limit to deter DOS attack
-        if (algorithm.equals("EC")) {
-            keySize = DEF_EC_KEY_SIZE;
-            if (minKeyLen < 112) {
-                minKeyLen = 112;
-            }
-            if (maxKeyLen > 2048) {
-                maxKeyLen = 2048;
+        int jdkMinKeyLen = 512;
+        int jdkMaxKeyLen = Integer.MAX_VALUE;
+        switch (algorithm) {
+            case "EC" -> {
+                keySize = DEF_EC_KEY_SIZE;
+                jdkMinKeyLen = 112;
+                jdkMaxKeyLen = 2048;
             }
-        } else {
-            if (algorithm.equals("DSA")) {
+            case "DSA" -> {
                 keySize = DEF_DSA_KEY_SIZE;
-            } else if (algorithm.equals("RSA")) {
+            }
+            case "RSA" -> {
                 keySize = DEF_RSA_KEY_SIZE;
-                if (maxKeyLen > 64 * 1024) {
-                    maxKeyLen = 64 * 1024;
-                }
-            } else {
+                jdkMaxKeyLen = 64 * 1024;
+            }
+            case "DH" -> {
                 keySize = DEF_DH_KEY_SIZE;
             }
-            if (minKeyLen < 512) {
-                minKeyLen = 512;
+            default -> {
+                throw new ProviderException
+                        ("Unrecognized algorithm for checking key size");
```

Comment:
If it's an unknown key algorithm, is it possible we just ignore it and keep using `minKeyLen` and `maxKeyLen`?

Reply:
Well, instead of ignore unknown key algorithm, perhaps safer to throw Exception so it can be caught and handled during develop time. P11KeyPairGenerator class is only used for known algorithms which it is registered for, so probably ok to go either way. I'd prefer to play it safe and force a review of this block of code when new algorithm is added.

Reply:
OK.